src/uproot/behaviors/TBranch.py

```diff
@@ -689,6 +689,229 @@ def foreach(start):
+
+            hasbranches.append(obj)
+
+            new_keys = obj.keys(
+                recursive=recursive,
+                filter_name=filter_name,
+                filter_typename=filter_typename,
+                filter_branch=real_filter_branch,
+                full_paths=full_paths,
+            )
+
+            if common_keys is None:
+                common_keys = new_keys
+            else:
+                new_keys = set(new_keys)
+                common_keys = [key for key in common_keys if key in new_keys]
+
+    if count == 0:
+        raise ValueError(
+            "allow_missing=True and no TTrees found in\n\n    {}".format(
+                "\n    ".join(
+                    "{"
+                    + "{}: {}".format(
+                        repr(f.file_path if isinstance(f, HasBranches) else f),
+                        repr(f.object_path if isinstance(f, HasBranches) else o),
+                    )
+                    + "}"
+                    for f, o in files
+                )
+            )
+        )
+
+    if len(common_keys) == 0 or not (all(is_self) or not any(is_self)):
+        raise ValueError(
+            "TTrees in\n\n    {}\n\nhave no TBranches in common".format(
+                "\n    ".join(
+                    "{"
+                    + "{}: {}".format(
+                        repr(f.file_path if isinstance(f, HasBranches) else f),
+                        repr(f.object_path if isinstance(f, HasBranches) else o),
+                    )
+                    + "}"
+                    for f, o in files
+                )
+            )
+        )
+
+    dask_dict = {}
+    for key in common_keys:
+        dask_arrays = []
+        for ttree in hasbranches:
+            entry_start, entry_stop = _regularize_entries_start_stop(
+                ttree.tree.num_entries, None, None
+            )
+            entry_step = 0
+            if uproot._util.isint(step_size):
+                entry_step = step_size
+            else:
+                entry_step = ttree.num_entries_for(step_size, expressions=f"{key}")
+
+            del_fn = dask.delayed(ttree[key].array)
+
+            dt = ttree[key].interpretation.numpy_dtype
+            if dt.subdtype is None:
+                inner_shape = ()
+            else:
+                dt, inner_shape = dt.subdtype
+
+            def foreach(start):
+                stop = min(start + entry_step, entry_stop)
+                length = stop - start
+
+                delayed_array = del_fn(library="np", entry_start=start, entry_stop=stop)
+                shape = (length,) + inner_shape
+                dask_arrays.append(
+                    da.from_delayed(delayed_array, shape=shape, dtype=dt)
+                )
+
+            # delayed_array = dask.delayed(ttree[key].array)(library="np")
+            # shape = (ttree[key].num_entries,) + inner_shape
+            # dask_arrays.append(da.from_delayed(delayed_array, shape=shape, dtype=dt))
```

Comment:
```suggestion
```

If this code is no longer relevant, it should be removed. (We have git for history; no need for it in comments.)